src/Microsoft.VisualBasic/tests/VB/FileIOTests.vb

```diff
@@ -0,0 +1,897 @@
+                Next
+                FileSystem.CopyDirectory(FullPathToSourceDirectory, FullPathToTargetDirectory, overwrite:=False)
+                Assert.Equal(IO.Directory.GetFiles(FullPathToSourceDirectory).Count, IO.Directory.GetFiles(FullPathToTargetDirectory).Count)
+                For Each CurrentFile As String In IO.Directory.GetFiles(FullPathToTargetDirectory)
+                    ' Ensure copy transferred written data
+                    Assert.True(HasExpectedData(CurrentFile, SourceData))
+                Next
+                IO.Directory.Delete(FullPathToTargetDirectory, recursive:=True)
+                IO.Directory.CreateDirectory(FullPathToTargetDirectory)
+                CreateTestFile(TestBase, DestData, PathFromBase:="TargetDirectory", TestFileName:=$"NewFile0")
+                Assert.Throws(Of IO.IOException)(Sub() FileSystem.CopyDirectory(FullPathToSourceDirectory, FullPathToTargetDirectory, overwrite:=False))
+                Assert.Equal(IO.Directory.GetFiles(FullPathToTargetDirectory).Count, IO.Directory.GetFiles(FullPathToSourceDirectory).Count)
+                For Each CurrentFile As String In IO.Directory.GetFiles(FullPathToTargetDirectory)
+                    If CurrentFile.EndsWith("0") Then
+                        ' Make sure file 0 is unchanged with DestDate
+                        Assert.True(HasExpectedData(CurrentFile, DestData))
+                    Else
+                        ' Ensure file 1 - 5 transferred SourData
+                        Assert.True(HasExpectedData(CurrentFile, SourceData))
+                    End If
+                Next
+            End Using
+        End Sub
+
+        <Fact>
+        Public Shared Sub CopyDirectory_SourceDirectoryName_DestinationDirectoryName_OverwriteTrue()
+            Using TestBase As New FileIOTests
+                Dim FullPathToSourceDirectory As String = IO.Path.Combine(TestBase.TestDirectory(), "SourceDirectory")
+                Dim FullPathToTargetDirectory As String = IO.Path.Combine(TestBase.TestDirectory(), "TargetDirectory")
+                IO.Directory.CreateDirectory(FullPathToSourceDirectory)
+                IO.Directory.CreateDirectory(FullPathToTargetDirectory)
+                For i As Integer = 0 To 5
+                    CreateTestFile(TestBase, SourceData, PathFromBase:="SourceDirectory", TestFileName:=$"NewFile{i}")
+                Next
+                FileSystem.CopyDirectory(FullPathToSourceDirectory, FullPathToTargetDirectory, overwrite:=True)
+                Assert.Equal(IO.Directory.GetFiles(FullPathToSourceDirectory).Count, IO.Directory.GetFiles(FullPathToTargetDirectory).Count)
+                For Each CurrentFile As String In IO.Directory.GetFiles(FullPathToTargetDirectory)
+                    ' Ensure copy transferred written data
+                    Assert.True(HasExpectedData(CurrentFile, SourceData))
+                Next
+            End Using
+        End Sub
```

Comment:
Is this test different than the first part of `CopyDirectory_SourceDirectoryName_DestinationDirectoryName` or `CopyDirectory_SourceDirectoryName_DestinationDirectoryName_OverwriteFalse`?